src/control/character_controller.rs

```diff
@@ -321,23 +322,23 @@ impl KinematicCharacterController {
                 break;
             }
 
-            result.grounded = self.detect_grounded_status_and_apply_friction(
-                dt,
-                bodies,
-                colliders,
-                queries,
-                character_shape,
-                &(Translation::from(result.translation) * character_pos),
-                &dims,
-                filter,
-                Some(&mut kinematic_friction_translation),
-                Some(&mut translation_remaining),
-            );
-
             if !self.slide {
                 break;
             }
         }
+
+        result.grounded = self.detect_grounded_status_and_apply_friction(
+            dt,
+            bodies,
+            colliders,
+            queries,
+            character_shape,
+            &(Translation::from(result.translation) * character_pos),
+            &dims,
+            filter,
+            Some(&mut kinematic_friction_translation),
+            Some(&mut translation_remaining),
+        );
```

Comment:
part 1 of the fix (the break from line 322 (in block `// No interference along the path.`) would bypass the ground detection, resulting in `not grounded`. That's happening because we're not checking for current colliding shapes. when doing the cast.



reworked from https://github.com/dimforge/rapier/pull/701/files#r1691673551

Reply:
after more testing with technique from https://github.com/dimforge/rapier/pull/705 ; this doesn't seem to improve significantly the behaviour, so I'll leave it out of this PR and focus on the slope. This PR builds a useful test to plus such verification into it though.